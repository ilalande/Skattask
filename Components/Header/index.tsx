import { useSelector } from 'react-redux';
import { signOut } from 'next-auth/react';
import { IRootState } from '@redux/reducers';
import Image from 'next/image';
import SHeader from './style';

export default function Header(): JSX.Element {
  //données récupérées du state, celui ci a été modifié par l'appel à l'action getLoggedUser depuis la page login
  const selectLoggedUser = useSelector(
    (state: IRootState) => state.users.loggedUser
  );

  return (
    <SHeader>
      <div className='containerGen'>
        <div className='title'>
          <Image
            src='/logo.png'
            width={120}
            height={50}
            layout='intrinsic'
            alt='todoList'
          />
        </div>
        <div className='button'>
          {selectLoggedUser && <span>{selectLoggedUser.name}</span>}
          <button
            onClick={() =>
              signOut({
                callbackUrl: 'http://localhost:3000/login',
              })
            }
          >
            sign out
          </button>
        </div>
      </div>
    </SHeader>
  );
}
