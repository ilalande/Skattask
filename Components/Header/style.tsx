import styled from 'styled-components';
import QUERIES from '../../styles/constant';

export default styled.header`
  min-height: 80px;
  padding-top: 0.5rem;
  margin-bottom: 0.5rem;

  .containerGen {
    display: flex;
    justify-content: end;
    align-items: center;
  }

  .title {
    min-width: 120px;
  }

  .button {
    display: block;

    width: max-content;
    margin: auto;
    margin-top: 10px;
  }

  .burgerMenu {
    display: none;
  }

  button {
    background-color: transparent;
    font-size: 0.8rem;
    text-transform: uppercase;
    border: 1px solid #e6e9ed;
    padding: 0.5rem 1rem;
    margin: 0 1rem;
  }
  @media ${QUERIES.mobileAndLess} {
    width: 100%;
    background-color: ${({ theme }) => theme.secondaryColor};

    .containerGen {
      width: 100%;
      justify-content: center;
    }
    .burgerMenu {
      display: block;

      width: fit-content;
      align-self: flex-start;
    }

    .burgerButton {
      display: block;
      border: 10px solid red;
      width: fit-content;
      padding: 0;
      margin: 0;
    }
  }
`;
