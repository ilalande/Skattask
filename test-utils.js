import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { mainTheme } from './styles/global-style';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom';
import 'jest-styled-components';

const initialState = {
  tasks: {
    allTasks: [],
  },
  task: {},
  users: { allUsers: [], loggedUser: {} },
};
const mockStore = configureStore();
const store = mockStore(initialState);

const AllTheProviders = ({ children }) => {
  return (
    <Provider store={store}>
      <ThemeProvider theme={mainTheme}>{children}</ThemeProvider>
    </Provider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';
export { customRender as renderWithProvider };

// import { render } from '@testing-library/react';
// import { ThemeProvider } from 'styled-components';
// import { mainTheme } from './styles/global-style';
// import configureStore from 'redux-mock-store';
// import { Provider } from 'react-redux';
// import '@testing-library/jest-dom';
// import 'jest-styled-components';
// import rootReducer from './redux/reducers/index';

// const initialState = {
//   tasks: {
//     allTasks: [],
//   },
//   task: {},
//   users: { allUsers: [], loggedUser: {} },
// };
// const mockStore = configureStore({
//   reducer: rootReducer,
//   initialState,
// });

// let options = {};
// options = { initialState, mockStore, ...options };

// const AllTheProviders = ({ children }) => {
//   return (
//     <Provider store={mockStore}>
//       <ThemeProvider theme={mainTheme}>{children}</ThemeProvider>
//     </Provider>
//   );
// };

// const customRender = (ui, options) =>
//   render(ui, { wrapper: AllTheProviders, ...options });

// // re-export everything
// export * from '@testing-library/react';
// export { customRender as renderWithProvider };
