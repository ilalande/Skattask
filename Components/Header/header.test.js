import { renderWithProvider, screen } from '../../test-utils';
import Header from '../Header';

import '@testing-library/jest-dom';

// From https://github.com/nextauthjs/next-auth/issues/4184

jest.mock('next-auth/react', () => {
  const originalModule = jest.requireActual('next-auth/react');
  const mockSession = {
    expires: new Date(Date.now() + 2 * 86400).toISOString(),
    user: { name: 'bob' },
  };
  return {
    __esModule: true,
    ...originalModule,
    useSession: jest.fn(() => {
      return { data: mockSession, status: 'unauthenticated' };
    }),
  };
});

describe('Header', () => {
  it('Matches snapshot', async () => {
    const header = renderWithProvider(<Header />);
    expect(header).toMatchSnapshot();
  });
  it('Shows "sign out" if user is logued', async () => {
    renderWithProvider(<Header />);
    const logoutButton = screen.getByRole('button');
    expect(logoutButton).toHaveTextContent('sign out');
  });
});
