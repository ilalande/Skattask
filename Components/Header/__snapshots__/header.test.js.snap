// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c0 {
  min-height: 80px;
  padding-top: 0.5rem;
}

.c0 .containerGen {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .title {
  min-width: 120px;
}

.c0 button {
  background-color: transparent;
  font-size: 0.8rem;
  text-transform: uppercase;
  border: 1px solid #e6e9ed;
  padding: 0.5rem 1rem;
  margin: 0 1rem;
}

<body>
    <div>
      <header
        class="c0"
      >
        <div
          class="containerGen"
        >
          <div
            class="title"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27120%27%20height=%2750%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="todoList"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
          <div
            class="button"
          >
            <span />
            <button>
              sign out
            </button>
          </div>
        </div>
      </header>
    </div>
  </body>,
  "container": .c0 {
  min-height: 80px;
  padding-top: 0.5rem;
}

.c0 .containerGen {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 .title {
  min-width: 120px;
}

.c0 button {
  background-color: transparent;
  font-size: 0.8rem;
  text-transform: uppercase;
  border: 1px solid #e6e9ed;
  padding: 0.5rem 1rem;
  margin: 0 1rem;
}

<div>
    <header
      class="c0"
    >
      <div
        class="containerGen"
      >
        <div
          class="title"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27120%27%20height=%2750%27/%3e"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt="todoList"
              data-nimg="intrinsic"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
        </div>
        <div
          class="button"
        >
          <span />
          <button>
            sign out
          </button>
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
